pkg/entity/register/worker_test.go

```diff
@@ -385,3 +401,161 @@ func TestWorker_registerEntitiesWithRetry_Success(t *testing.T) {
+				Entity: entity.Fields{Name: "warnings"},
+			},
+		},
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-reqsRegisteredQueue:
+				// empty the registered queue to process new requests.
+			case <-time.After(5 * time.Second):
+				cancel()
+				t.Error("Timeout while executing the test")
+			}
+		}
+	}()
+
+	hook := new(test.Hook)
+	log.AddHook(hook)
+	log.SetOutput(ioutil.Discard)
+
+	batchSizeBytes := 10000
+	w.send(ctx, batch, &batchSizeBytes)
+
+	searchLogEntries := func(expectedMessages []string, level log.Level) (found bool) {
+		for _, expectedMsg := range expectedMessages {
+			for i, entry := range hook.AllEntries() {
+				if entry.Level != level {
+					continue
+				}
+				if val, ok := hook.AllEntries()[i].Data["error"]; ok {
+					errStr := val.(error).Error()
+					if errStr == expectedMsg {
+						found = true
+					}
+				}
+			}
+		}
+		return
+	}
+
+	assert.Eventually(t, func() bool {
+		ok := searchLogEntries(expectedErrs, log.ErrorLevel)
+		return ok
+	}, time.Second, 10*time.Millisecond,
+		"expected to find error messages: %s", expectedErrs)
+
+	assert.Eventually(t, func() bool {
+		ok := searchLogEntries(expectedWarnings, log.WarnLevel)
+		return ok
+	}, time.Second, 10*time.Millisecond,
+		"expected to find warning messages: %s \n", expectedWarnings)
+}
+
+func TestWorker_send_Logging_VerboseDisabled(t *testing.T) {
+	// When the request is successful but some entities fail, we log only when verbose is enabled.
+	reqsToRegisterQueue := make(chan fwrequest.EntityFwRequest, 0)
+	reqsRegisteredQueue := make(chan fwrequest.EntityFwRequest, 0)
+
+	agentIdentity := func() entity.Identity {
+		return entity.Identity{ID: 12}
+	}
+
+	client := &fakeClient{
+		ids: []entity.ID{13, 14},
+		// no err from backend.
+		err: nil,
+	}
+
+	config := WorkerConfig{
+		VerboseLogLevel: 0,
+	}
+	w := NewWorker(agentIdentity, client, backoff.NewDefaultBackoff(), reqsToRegisterQueue, reqsRegisteredQueue, config)
+
+	batch := map[entity.Key]fwrequest.EntityFwRequest{
+		entity.Key("error"): {
+			Data: protocol.Dataset{
+				Entity: entity.Fields{Name: "error"},
+			},
+		},
+		entity.Key("warnings"): {
+			Data: protocol.Dataset{
+				Entity: entity.Fields{Name: "warnings"},
+			},
+		},
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-reqsRegisteredQueue:
+				// empty the registered queue to process new requests.
+			case <-time.After(5 * time.Second):
+				cancel()
+				t.Error("Timeout while executing the test")
+			}
+		}
+	}()
+
+	hook := new(test.Hook)
+	log.AddHook(hook)
+	log.SetOutput(ioutil.Discard)
+
+	batchSizeBytes := 10000
+	w.send(ctx, batch, &batchSizeBytes)
+
+	assert.Eventually(t, func() bool {
+		hook.LastEntry()
+		return len(hook.Entries) == 0
+	}, time.Second, 10*time.Millisecond)
```

Comment:
Sorry, my bad. I thought you were checking that it had an item. You can just replace the eventually with the following:
```suggestion
	assert.Empty(t, hook.AllEntries())
```

Basically calling `hook.Entries` is what causes race conditions and flaky tests. It is better to call `hook.AllEntries()` as that is thread safe.